packages/services/src/kernel/kernel.ts

```diff
@@ -896,6 +941,21 @@ export interface IComm extends IDisposable {
    */
   readonly targetName: string;
 
+  /**
+   * The subshell ID, null if not using a subshell.
+   */
+  subshellId: string | null;
+
+  /**
+   * Promise that resolves when the subshell started, if any
+   */
+  get subshellStarted(): Promise<void>;
```

Comment:
@martinRenou to highlight, `subshellStarted` and `subshellId` are also non-optional additions.

Reply:
yeah I'm annoyed to have added this. But this is needed for testing

Reply:
Could casting in tests to `as CommHandler` make the tests pass while narrowing the required API?

Reply:
What do you mean by narrowing the required API?

Reply:
Gotcha, I misunderstood the casting to `as CommHandler` in tests. Let me try to do it.

Reply:
Done!